x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlEntityDescriptorBuilder.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.authc.saml;
+
+import org.elasticsearch.cli.ExitCodes;
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.cli.UserException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.xpack.core.security.authc.RealmConfig;
+import org.elasticsearch.xpack.core.security.authc.RealmSettings;
+import org.elasticsearch.xpack.core.security.authc.saml.SamlRealmSettings;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.saml.saml2.metadata.EntityDescriptor;
+
+import java.util.Locale;
+import java.util.Map;
+
+public class SamlEntityDescriptorBuilder {
+    private final EntityDescriptor entityDescriptor;
+
+    public EntityDescriptor getEntityDescriptor() {
+        return entityDescriptor;
+    }
+
+    public SamlEntityDescriptorBuilder(SamlRealm samlRealm) throws Exception{
+        final SpConfiguration spConfig = samlRealm.getLogoutHandler().getSpConfiguration();
+        final SamlSpMetadataBuilder builder = new SamlSpMetadataBuilder(samlRealm)
+            .encryptionCredentials(spConfig.getEncryptionCredentials())
+            .signingCredential(spConfig.getSigningConfiguration().getCredential())
+            .authnRequestsSigned(spConfig.getSigningConfiguration().shouldSign(AuthnRequest.DEFAULT_ELEMENT_LOCAL_NAME));
+        entityDescriptor = builder.build();
+    }
+
+    public SamlEntityDescriptorBuilder(RealmConfig realm, boolean batch, String serviceName, String orgName, String orgUrl,
```

Comment:
This is probably irrelevant now, if we remove this class after all but it wouldn't make much sense for user input related code to be here and not in the `Command` class

Reply:
Refactored